packages/legend-query-builder/src/components/watermark/QueryBuilderWatermark.tsx

```diff
@@ -0,0 +1,213 @@
+              >
+                <BasicValueSpecificationEditor
+                  valueSpecification={guaranteeNonNullable(
+                    watermarkState.value,
+                  )}
+                  setValueSpecification={(val: ValueSpecification): void => {
+                    watermarkState.setValue(val);
+                  }}
+                  graph={graph}
+                  obseverContext={
+                    watermarkState.queryBuilderState.observableContext
+                  }
+                  typeCheckOption={{
+                    expectedType: graph.getPrimitiveType(PRIMITIVE_TYPE.STRING),
+                    match:
+                      parameterType ===
+                      graph.getPrimitiveType(PRIMITIVE_TYPE.STRING),
+                  }}
+                  resetValue={(): void => watermarkState.setValue(undefined)}
+                />
+              </PanelDropZone>
+            </div>
+          </div>
+        )}
+      </>
+    );
+  },
+);
+
+export const QueryBuilderWatermarkEditor = observer(
+  (props: { queryBuilderState: QueryBuilderState }) => {
+    const { queryBuilderState } = props;
+
+    const watermarkState = queryBuilderState.watermarkState;
+
+    const handleClose = (): void => {
+      if (watermarkState.value instanceof PrimitiveInstanceValue) {
+        if (watermarkState.value.values.length < 1) {
+          watermarkState.setValue(undefined);
+        } else {
+          if ((watermarkState.value.values[0] as string).length === 0) {
+            watermarkState.setValue(undefined);
+          }
+        }
+      }
+      queryBuilderState.setIsEditingWatermark(false);
+    };
+
+    return (
+      <Dialog
+        open={Boolean(queryBuilderState.isEditingWatermark)}
+        onClose={handleClose}
+        classes={{
+          root: 'editor-modal__root-container',
+          container: 'editor-modal__container',
+          paper: 'editor-modal__content',
+        }}
+      >
+        <Modal darkMode={true} className="editor-modal">
+          <ModalHeader title="Watermark" />
+          <ModalBody>
+            <PanelForm>
+              <PanelFormBooleanField
+                isReadOnly={false}
+                value={watermarkState.value !== undefined}
+                prompt="Add a watermark"
+                update={(): void => watermarkState.toogleWatermark()}
+              />
+              {watermarkState.value && (
+                <>
+                  <WatermarkValueEditor
+                    watermarkState={watermarkState}
+                    queryBuilderState={queryBuilderState}
+                  />
+                  <PanelDivider />
+                  <PanelFormListItems title="List of available parameters">
+                    {queryBuilderState.parametersState.parameterStates
+                      .length === 0 && <> No parameters in query </>}
+                    {queryBuilderState.parametersState.parameterStates.map(
```

Comment:
As mentioned, lets isolate the parameters we show as only parameters that meet the input of the `forWatermark` function. 
`meta::datalake::functions::forWatermark<T>(collection: T[*], watermak: String[1]): T{*]` as seen here the watermark value is of type string with multiplicity 1.

Reply:
makes sense! 🙇